### ISSUE TITLE
Support for PATCH http method

### ISSUE THREAD

**Issue by Rydgel** [finagle/featherbed#50]
It seems like I can only do get, post, put, delete or head requests.
In the current project I'm working on, I will need the support for the PATCH request.

Do you think it can be added easily?

**Comment by jeremyrsmith** [comment]
@Rydgel yes, we should absolutely support `PATCH`. I'm not sure if it was an oversight or if I was deferring this until settling on a good representation of a data type for abstracting over `PATCH`. The conventions around `PATCH` are not all that straightforward, so it's a bit of a minefield.
